src/index.js

```diff
@@ -123,49 +124,53 @@ export default declare(({
   return {
     visitor: {
       Program: {
-        enter(path, { file, opts, filename }) {
+        enter(rootPath, state) {
+          const { opts, filename, file } = state;
           if (typeof filename === 'string' && typeof opts.filename !== 'undefined') {
             throw new TypeError('the "filename" option may only be provided when transforming code');
           }
           if (typeof filename === 'undefined' && typeof opts.filename !== 'string') {
             throw new TypeError('the "filename" option is required when transforming code');
           }
-          if (!path.scope.hasBinding('React')) {
+
+          if (!opts.noReactAutoImport && !rootPath.scope.hasBinding('React')) {
```

Comment:
Can you help me understand why this line alone is not sufficient to address the issue?

Reply:
The `applyPlugin` calls need to be made before subsequent plugins (specifically the `next/babel` preset) are called. The previous method called the JSX transform's `visitor.Program.enter` before doing those, causing the JSX transform to incorrectly think that the file didn't contain any JSX.

Reply:
Transform ordering is something that is usually dealt with in the babel config. This does mean, if you need a plugin to run before a preset, that you have to create your own preset that uses the plugin, and then use that preset in your own config instead of that plugin.

If you do that, does that obviate the need for all the larger changes?

Reply:
Yes, but that would require a lot of work for anyone that uses presets. I think it makes more sense to ensure the plugin runs before everything else does. I believe that other plugins in the `plugins` array are also run before `applyPlugin`.

Reply:
That's the nature of babel, though - and not something individual transforms should be solving.

Reply:
How would you suggest approaching adding compatibility for the React JSX transform? The transform needs to see JSX in the file in order to execute.

Reply:
I'd assume that those using it need to ensure that *it* runs last.

Reply:
The thing is, it can never run last if the visitor isn't written this way. `visitor.Program.enter` of all plugins is run before we ever hit an `applyPlugin`, and the React JSX transform's `visitor.Program.enter` determines whether or not it will operate on the file.

Reply:
aha, ok thanks for clarifying - iow, we must write it this way explicitly because of the way the new jsx transform is authored.

Might this cause an ordering change that could break existing users?

Reply:
It's possible, since subsequent plugins will now be seeing the SVGs as JSX instead of as imports

Reply:
Could we make the ordering changes pivot based on the `noReactAutoImport` setting?